torchgeo/models/scale_mae.py

```diff
@@ -0,0 +1,252 @@
+        )
+        cls_tokens = self.cls_token.expand(x.shape[0], -1, -1)
+        x = torch.cat((cls_tokens, x), dim=1)
+        x = x + pos_embed
+        x = self.pos_drop(x)
+        return x
+
+
+def interpolate_pos_embed(
+    model: ScaleMAEViT, state_dict: OrderedDict[str, Tensor]
+) -> OrderedDict[str, Tensor]:
+    """Interpolate the positional embeddings if image size is different than pretrained image size.
+
+    Args:
+        model: ScaleMAEViT model.
+        state_dict: Pretrained model state dict.
+
+    Returns:
+        state_dict: State dict with interpolated positional embeddings.
+    """
+    pos_embed_checkpoint = state_dict['pos_embed']
+    embedding_size = pos_embed_checkpoint.shape[-1]
+    num_patches = model.patch_embed.num_patches
+    num_extra_tokens = model.pos_embed.shape[-2] - num_patches
+    # height (== width) for the checkpoint position embedding
+    orig_size = int((pos_embed_checkpoint.shape[-2] - num_extra_tokens) ** 0.5)
+    # height (== width) for the new position embedding
+    new_size = int(num_patches**0.5)
+    # class_token and dist_token are kept unchanged
+    if orig_size != new_size:
+        print(
+            'Interpolating positional embeddings from %dx%d to %dx%d'
+            % (orig_size, orig_size, new_size, new_size)
+        )
+        extra_tokens = pos_embed_checkpoint[:, :num_extra_tokens]
+        # only the position tokens are interpolated
+        pos_tokens = pos_embed_checkpoint[:, num_extra_tokens:]
+        pos_tokens = pos_tokens.reshape(
+            -1, orig_size, orig_size, embedding_size
+        ).permute(0, 3, 1, 2)
+        pos_tokens = torch.nn.functional.interpolate(
+            pos_tokens, size=(new_size, new_size), mode='bicubic', align_corners=False
+        )
+        pos_tokens = pos_tokens.permute(0, 2, 3, 1).flatten(1, 2)
+        new_pos_embed = torch.cat((extra_tokens, pos_tokens), dim=1)
+        state_dict['pos_embed'] = new_pos_embed
+
+    return state_dict
+
+
+# https://github.com/pytorch/vision/pull/6883
+# https://github.com/pytorch/vision/pull/7107
+# Can be removed once torchvision>=0.15 is required
+Weights.__deepcopy__ = lambda *args, **kwargs: args[0]
+
+
+class ScaleMAE_ViTLarge16_Weights(WeightsEnum):  # type: ignore[misc]
+    """Scale-MAE ViT Large patch size 16 weights.
+
+    .. versionadded:: 0.6
+    """
+
+    FMOW_RGB = Weights(
+        url='https://hf.co/torchgeo/vit_large_patch16_224_fmow_rgb_scalemae/resolve/9dc7f569424baeb780698352cf6e87638c882123/vit_large_patch16_224_fmow_rgb_scalemae-98ed9821.pth',  # noqa: E501
+        transforms=_scale_mae_transforms,
+        meta={
+            'dataset': 'fMoW',
+            'in_chans': 3,
+            'img_size': 224,
+            'model': 'vit_large_patch16',
+            'publication': 'https://arxiv.org/abs/2212.14532',
+            'repo': 'https://github.com/bair-climate-initiative/scale-mae',
+        },
+    )
+
+
+def scalemae_vit_large_patch16(
```

Comment:
```suggestion
def scalemae_large_patch16_224(
```

Reply:
Or is the image size customizable?

Reply:
Can we add additional sizes? I know we only have pre-trained weights available for large, but we might as well add functions to instantiate other sizes like they do in the source repo.

Reply:
The image size can be changed and the positional embeddings will get interpolated.

Reply:
Updated diff to remove 244